### PR TITLE
[bitnami/kong] Adapt Kong tests to Air-gapped environments

### PR DESCRIPTION
**Pull request by jotamartos**
### Description of the change

Some tests do not work in air-gapped environments.

### Benefits

We are modifying the tests to make them work in all environments. This way we can verify that the solution works as expected. If the solution doesn't work in that environment, the tests will not be changed and support for that env will be removed.

### Applicable issues

- fixes #

### Checklist

- [NA] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [NA] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/kong/cypress/cypress/integration/kong_spec.js [hunk 1]
```diff
@@ -51,6 +51,14 @@ it('allows adding a new service and route through admin API', () => {
       expect(response.status).to.eq(201);
     });
 
+    cy.request({
+      method: 'GET',
+      url: `${BASE_URL}/services`,
```
Comment (FraPazGal) [inline .vib/kong/cypress/cypress/integration/kong_spec.js:56]: IIRC, we shouldn't need to add `${BASE_URL}` here, as Cypress already does so by default.

File: .vib/kong/cypress/cypress/integration/kong_spec.js [hunk 2]
```diff
@@ -51,6 +51,14 @@ it('allows adding a new service and route through admin API', () => {
...
+    }).then((response) => {
+      expect(response.status).to.eq(200);
+      expect(JSON.stringify(response.body)).to.contain(services.newService.upstreamURL.replace(/^https?:\/\//, ''));
```
Comment (FraPazGal) [inline .vib/kong/cypress/cypress/integration/kong_spec.js:59]: Not really an issue but just a FYI, I'll say there is no need to use `JSON.stringify()` for the response's body.

File: .vib/kong/cypress/cypress/integration/kong_spec.js [hunk 3]
```diff
@@ -68,10 +76,10 @@ it('allows adding a new service and route through admin API', () => {
       cy.wait(7000);
       cy.request({
         method: 'GET',
-        url: `${routes.newRoute.path}${random}`,
+        url: `${BASE_URL}/services/${services.newService.name}-${random}/routes`,
       }).then((response) => {
         expect(response.status).to.eq(200);
-        expect(response.body).to.contain(services.newService.upstreamContent);
+        expect(JSON.stringify(response.body)).to.contain(`${routes.newRoute.path}${random}`);
```
Comment (FraPazGal) [inline .vib/kong/cypress/cypress/integration/kong_spec.js:82]: We should remove `services.newService.upstreamContent` from the `services.json` file if we no longer use it.

**Review by FraPazGal — COMMENTED** [review]
Thanks for the PR @jotamartos, I have added a few comments as a review. Please check them whenever you can

File: .vib/kong/cypress/cypress/integration/kong_spec.js (hunk 2 above)
Comment (jotamartos) [reply .vib/kong/cypress/cypress/integration/kong_spec.js:59]: The response is an object and we need to make it a string for this to work.

File: .vib/kong/cypress/cypress/integration/kong_spec.js (hunk 3 above)
Comment (jotamartos) [reply .vib/kong/cypress/cypress/integration/kong_spec.js:82]: You are right, will do it!

File: .vib/kong/cypress/cypress/integration/kong_spec.js (hunk 1 above)
Comment (jotamartos) [reply .vib/kong/cypress/cypress/integration/kong_spec.js:56]: There are 2 endpoints here, the proxy and the API. Will investigate why it's needed but if I'm not mistaken, it's necessary for the test to work.

Comment (jotamartos) [reply .vib/kong/cypress/cypress/integration/kong_spec.js:56]: BASE_URL is set some lines above to include the port of the API endpoint.

**Comment by jotamartos** [comment]
Tests succeeded here: https://github.com/bitnami/charts/actions/runs/5691123284

**Pull request merged**